### PR TITLE
feat(collector-landing-page): resolve loading of p-data-loader at landing page

### PR DESCRIPTION
**Pull request by skdud4659**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
- Resolve loading of `p-data-loader` at collector landing page

**KEYPOINT**
When the page is rendered, we check the 'hasCollectorList' value through the collector list API to determine whether it has been filtered by the provider or not.

1. When there is no collector list data.
commit: b8e38dc
image

---

2. When there is no collector list based on the selected provider + apply contents loader.
commit: 0e3e1985203187b6a9056298ad8238a7c20bebb4
image

### Things to Talk About

🐱: The content and code I submitted in the previous PR are almost similar, but there have been some changes in the positioning and a few additions to apply the page's initial entry and content loader.
Please check changes by commit!

### CODE REVIEW

File: apps/web/src/services/asset-inventory/store/collector-page-store.ts [hunk 1]
```diff
@@ -30,14 +29,14 @@ export const useCollectorPageStore = defineStore('collector-page', {
...
-        async getCollectorList(queryData) {
+        async getCollectorList(queryData?: Query) {
             this.loading = true;
             try {
                 const res = await SpaceConnector.client.inventory.collector.list({
                     query: queryData,
                 });
-                this.totalCount = res.total_count;
                 this.collectors = res.results;
+                this.listCount = res.total_count;
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/store/collector-page-store.ts:39]: Question: Why did you change the variable from `totalCount` to `listCount`? I think `totalCount` is easier to understand.

File: apps/web/src/services/asset-inventory/collector/type.ts [hunk 2]
```diff
@@ -145,3 +145,26 @@ export interface ScheduleGetParameter {
...
+interface CollectorPlugin {
+    name?: string;
+    icon?: string;
+    info: CollectorPluginModel
+}
+interface CollectorDetailLink {
+    name: string;
+    param: CollectorDetailLinkParameter,
+    query: CollectorDetailLinkQuery
+}
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/type.ts:158]: nitpick:

```typescript
interface CollectorDetailLink {
    name: string;
    param: CollectorDetailLinkParameter;
    query: CollectorDetailLinkQuery;
}
```

File: apps/web/src/services/asset-inventory/collector/type.ts [hunk 3]
```diff
@@ -145,3 +145,26 @@ export interface ScheduleGetParameter {
...
+interface CollectorDetailLinkQuery {
+    filters: string[]
+}
+export interface CollectorItemInfo {
+    collectorId: string,
+    name: string,
+    plugin: CollectorPlugin,
+    detailLink: CollectorDetailLink
+}
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/type.ts:170]: nitpick: semicolon!

**Review by yuda110 — APPROVED** [review]
Nice! I left some comments :)

File: apps/web/src/services/asset-inventory/collector/modules/CollectorContents.vue [hunk 4]
```diff
@@ -1,65 +1,56 @@
 <template>
     <div class="collector-contents">
-        <provider-list
-            :provider-list="props.providerList"
-            :selected-provider="state.selectedProvider"
-            @change-provider="handleSelectedProvider"
-        />
```
Comment (sulmoJ) [inline apps/web/src/services/asset-inventory/collector/modules/CollectorContents.vue:7]: Praise: I love this separation!

File: apps/web/src/services/asset-inventory/collector/CollectorPage.vue [hunk 5]
```diff
@@ -134,27 +110,35 @@ const collectorApiQueryHelper = new ApiQueryHelper()
...
 };
 
-/* Watcher */
-watch(() => collectorPageState.selectedProvider, async () => {
-    await initCollectorList();
-});
-
 /* Unmounted */
 onUnmounted(() => {
     collectorPageStore.$reset();
```
Comment (sulmoJ) [inline apps/web/src/services/asset-inventory/collector/CollectorPage.vue:144]:
```suggestion
    collectorPageStore.$dispose();
```
suggestion: I recommend using `$dispose()` because it completely removes the store. You can also confirm its complete removal in the dev tools.
(`$reset()` is included in`$dispose`)

**Review by sulmoJ — APPROVED** [review]
In the PR description, I mistook Commit ID for a color palette! `#xxxxxxx` How about writing `commit: xxxxxxx` instead?
image

LGTM!!
